mod/node-api/backend/block.go

```diff
@@ -46,26 +45,21 @@ func (b Backend[
 // GetBlockRoot returns the root of the block at the given stateID.
 func (b Backend[
 	_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _,
-]) BlockRootAtSlot(
-	slot uint64,
-) (common.Root, error) {
+]) BlockRootAtSlot(slot math.Slot) (common.Root, error) {
 	st, slot, err := b.stateFromSlot(slot)
 	if err != nil {
 		return common.Root{}, err
 	}
 
 	// As calculated by the beacon chain. Ideally, this logic
 	// should be abstracted by the beacon chain.
-	index := slot % b.cs.SlotsPerHistoricalRoot()
-	return st.GetBlockRootAtIndex(index)
+	return st.GetBlockRootAtIndex(slot.Unwrap() % b.cs.SlotsPerHistoricalRoot())
```

Comment:
this Unwrap is needed since chain spec returns uint64. this unwrap also is done in every StateProcesso for every slot.